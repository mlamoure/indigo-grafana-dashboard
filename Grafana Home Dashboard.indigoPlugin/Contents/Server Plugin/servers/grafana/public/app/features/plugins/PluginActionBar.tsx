import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import LayoutSelector, { LayoutMode } from '../../core/components/LayoutSelector/LayoutSelector';
import { setLayoutMode, setPluginsSearchQuery } from './state/actions';
import { getPluginsSearchQuery, getLayoutMode } from './state/selectors';

export interface Props {
  searchQuery: string;
  layoutMode: LayoutMode;
  setLayoutMode: typeof setLayoutMode;
  setPluginsSearchQuery: typeof setPluginsSearchQuery;
}

export class PluginActionBar extends PureComponent<Props> {
  onSearchQueryChange = event => {
    this.props.setPluginsSearchQuery(event.target.value);
  };

  render() {
    const { searchQuery, layoutMode, setLayoutMode } = this.props;

    return (
      <div className="page-action-bar">
        <div className="gf-form gf-form--grow">
          <label className="gf-form--has-input-icon">
            <input
              type="text"
              className="gf-form-input width-20"
              value={searchQuery}
              onChange={this.onSearchQueryChange}
              placeholder="Filter by name or type"
            />
            <i className="gf-form-input-icon fa fa-search" />
          </label>
          <LayoutSelector mode={layoutMode} onLayoutModeChanged={(mode: LayoutMode) => setLayoutMode(mode)} />
        </div>
        <div className="page-action-bar__spacer" />
        <a
          className="btn btn-success"
          href="https://grafana.com/plugins?utm_source=grafana_plugin_list"
          target="_blank"
        >
          Find more plugins on Grafana.com
        </a>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    searchQuery: getPluginsSearchQuery(state.plugins),
    layoutMode: getLayoutMode(state.plugins),
  };
}

const mapDispatchToProps = {
  setPluginsSearchQuery,
  setLayoutMode,
};

export default connect(mapStateToProps, mapDispatchToProps)(PluginActionBar);
