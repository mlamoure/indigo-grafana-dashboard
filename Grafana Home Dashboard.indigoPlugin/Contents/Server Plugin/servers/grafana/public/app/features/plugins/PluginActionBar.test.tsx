import React from 'react';
import { shallow } from 'enzyme';
import { PluginActionBar, Props } from './PluginActionBar';
import { LayoutModes } from '../../core/components/LayoutSelector/LayoutSelector';

const setup = (propOverrides?: object) => {
  const props: Props = {
    searchQuery: '',
    layoutMode: LayoutModes.Grid,
    setLayoutMode: jest.fn(),
    setPluginsSearchQuery: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<PluginActionBar {...props} />);
  const instance = wrapper.instance() as PluginActionBar;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
