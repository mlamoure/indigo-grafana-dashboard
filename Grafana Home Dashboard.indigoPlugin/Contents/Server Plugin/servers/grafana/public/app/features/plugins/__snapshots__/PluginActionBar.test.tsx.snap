// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <label
      className="gf-form--has-input-icon"
    >
      <input
        className="gf-form-input width-20"
        onChange={[Function]}
        placeholder="Filter by name or type"
        type="text"
        value=""
      />
      <i
        className="gf-form-input-icon fa fa-search"
      />
    </label>
    <LayoutSelector
      mode="grid"
      onLayoutModeChanged={[Function]}
    />
  </div>
  <div
    className="page-action-bar__spacer"
  />
  <a
    className="btn btn-success"
    href="https://grafana.com/plugins?utm_source=grafana_plugin_list"
    target="_blank"
  >
    Find more plugins on Grafana.com
  </a>
</div>
`;
